src/vs/workbench/parts/terminal/electron-browser/windowsShellHelper.ts

```diff
@@ -13,9 +13,20 @@ import XTermTerminal = require('xterm');
 
 const SHELL_EXECUTABLES = ['cmd.exe', 'powershell.exe', 'bash.exe'];
 
+interface IWindowsTerminalProcess {
+	pid: number;
+	executable: string;
+}
+
+interface IWindowsProcTreeItem {
+	pid: number;
+	children: IWindowsTerminalProcess[];
+}
+
 export class WindowsShellHelper {
-	private _childProcessIdStack: number[];
+	private _childProcessIdStack: IWindowsProcTreeItem[];
```

Comment:
Perhaps it would be easier/nicer to store the processes in a tree instead of an array now? ie.

```
interface IWindowsProcTreeItem {
	pid: number;
	children: IWindowsProcTreeItem[];
}
```